protoc-gen-pbandk/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -507,7 +508,12 @@ public open class CodeGenerator(
     protected val File.Type.Message.mapEntryValueKotlinType: String?
         get() = if (!mapEntry) null else (fields[1] as File.Field.Numbered.Standard).kotlinValueType(true)
 
-    protected fun File.Field.Numbered.kotlinValueType(allowNulls: Boolean): String = when (this) {
+    protected val File.Field.Numbered.allowNulls: Boolean
+        get() =  requiredFieldNumber?.let {
+            // 1 means `required = true` so we can't allow nulls
+            this.options.unknownFields[it]?.values?.get(0)?.rawBytes?.array?.get(0)?.toInt() != 1
+        } ?: true
+    protected fun File.Field.Numbered.kotlinValueType(allowNulls: Boolean = this.allowNulls): String = when (this) {
```

Comment:
what do you think about putting something like 

```
val isRequired: Boolean
            get() = rawBytes.array.first().toInt() != 1
```

in the `UnknownField.Value` class and then having 

`options.unknownFields[it]?.values?.firstOrNull()?.isRequired == true` 

here?

Reply:
Maybe not `isRequired` because it is just a boolean. We need to smart cast those bytes to a boolean, maybe something like `asBoolean()` is a better candidate.